### PR TITLE
Add Time comparison for KD_TREE. 

### PR DESCRIPTION
**Pull request by sudk1896**
This adds Time comparison for `KD_TREE`. #3513 
@karlnapf : I had an older version of IPython on my machine (hence so many edits). Take a look now.

### CODE REVIEW

**Comment by karlnapf** [comment]
Cool thanks!
Can you post a gist of the notebook with outputs?

**Comment by sudk1896** [comment]
@karlnapf : The gist is [here](https://gist.github.com/sudk1896/83b90028a0bd6e0de6c9bd6c45e5f80a)
I haven't uploaded the whole notebook, just the part where I made the changes, specifically the time comparison for `KD_TREE`. The time for all three solvers is there.

**Comment by karlnapf** [comment]
Can you please post the full notebook so I can see how it looks?
Reading the text changes is a bit time consuming in diff otherwise

**Comment by sudk1896** [comment]
@karlnapf: The whole notebook as a gist is [here](https://gist.github.com/sudk1896/99aa1a17c4ed6a735756356c96bce267).

File: doc/ipython-notebooks/multiclass/KNN.ipynb [hunk 1]
```diff
@@ -230,14 +230,14 @@
...
+    "Obviously applying KNN is very costly: for each prediction you have to compare the object against all training objects. While the implementation in `SHOGUN` will use all available CPU cores to parallelize this computation it might still be slow when you have big data sets. In `SHOGUN`, you can use *Cover Trees* to speed up the nearest neighbor searching process in KNN. Just call `set_use_covertree` on the KNN machine to enable or disable this feature. `SHOGUN` also has a third KNN solver called `KD-Trees`, It is faster than the other two solvers only on low-dimesional data. We also show the prediction time comparison between all the three solvers in this tutorial. So let's just have a comparison utilizing the data above:"
```
Comment (karlnapf) [inline doc/ipython-notebooks/multiclass/KNN.ipynb]: typo "It is " -> "it is"  ... even better would be "which is"

**Comment by karlnapf** [comment]
So you didn't add the KDtree to the last comparison?

**Comment by sudk1896** [comment]
@karlnapf: I did (Earlier there were comparisons only for the two solvers). Now there is for the third one i.e. `KD-Tree`.

**Comment by karlnapf** [comment]
No you did not, *as I wrote above* I mean the *last* comparison.

The change you did is useful, but it would be cooler to apply it to the whole notebook you know

**Comment by karlnapf** [comment]
closing for now. Feel free to re-open if you put updates

**Pull request closed without merging**